docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
+
+### Proposal A
+
+The communication of cloud edge components/services is realized based on L4/L7 proxy, which is similar to kubedge's cloud-edge message channel.
+
+Cloudgateway and edgegateway are introduced into the central cloud and each edge site respectively. The edge site edgegateway supports connecting to the cloud gateway of the central cloud when it is started. The service communication between cloud and edge needs to go through these two gateways before arriving. Service access interfaces are provided on these two gateways to receive requests and route requests to target services.
+
+* Cloudgateway and edgegateway are installed and run in the central cloud and each edge site respectively
+* Edgegateway of Edge site can actively establish encrypted links with cloudgateway on the central side, and support two-way message transmission
+* Register the services of edge site and central cloud to cloudgateway
+* Edgegateway in the edge site acts as the service in the central cloud, and exposes the service access interface of the central cloud to the service in the edge site
+* Cloudgateway in the central cloud acts as the service in each edge site and exposes the service access interface of the edge site in the central cloud
+
+### Proposal B
+
+The access of cloud edge components/services is realized based on L3 layer network communication, but the network/IP is not exposed to the remote end, but the components/services are exposed to the remote end.
+
+![](Proposal-B1.png)
+
+Consider providing the concept of virtual presence, which is the local virtual presentation of remote services. In this way, virtual presence can be regarded as a local service with local IP addresses. These IP addresses have nothing to do with the network where the remote service is located, and can be allocated independently. In this way, the problem of unified planning of IP address of VPN solution in the whole network can be solved at the network level. From the perspective of security, virtual presence is completely isolated from the remote service. The client cannot directly access the network where the remote service is located. If the client is attacked, it can not directly attack the server.
+
+![](Proposal-B2.png)
+
+Gateway (cloudgateway/edgegateway) is introduced into the cloud and each edge. Virtual presence of remote service is provided on these gateways, and service components that need communication between cloud and edge site are presented respectively to realize communication access of cloud edge service：
+
+* Edgegateway is installed and run in the central cloud and each edge site respectively, so that these edge sites can access to the central cloud through edgegateway
+* Create edgesite objects to the system and associate edgesite with these edgegateway
+* All components in the cloud and edge that need mutual access are registered with the central cloud cloudgateway in the form of service, and the access policies of these services are configured
```

Comment:
Here it is said that  cloud gateway is used in central cloud. I am confused.

Reply:
Yes, the cloudgateway is running in the cloud, and the edgegateway is running in the edge(docs/design-proposals/cloud-edge-communication/Proposal-B2.png)